test/jdk/java/lang/constant/MethodHandleDescTest.java

```diff
@@ -359,7 +361,21 @@ public void testSymbolicDescsConstants() throws ReflectiveOperationException {
 
     public void testKind() {
         for (Kind k : Kind.values()) {
+            assertEquals(Kind.valueOf(k.refKind), Kind.valueOf(k.refKind, k.refKind == MethodHandleInfo.REF_invokeInterface));
```

Comment:
@mlchung mlchung 2 days ago Member

Looks like the test does not verify the cases specified by valueOf(int refKind, boolean isInterface).
i.e. For most values of refKind, there is an exact match regardless of the value of isInterface except REF_invokeStatic and REF_invokeSpecial.

Do you mind adding those cases?
@vicente-romero-oracle vicente-romero-oracle 22 hours ago •
Author Member

    Looks like the test does not verify the cases specified by valueOf(int refKind, boolean isInterface).
    i.e. For most values of refKind, there is an exact match regardless of the value of isInterface except REF_invokeStatic and REF_invokeSpecial.

    Do you mind adding those cases?

hum, the implementation for valueOf(int refKind, boolean isInterface) is incorrect, the behavior does depends on the value of isInterface for example: Kind.valueOf(1, false) returns GETTER while Kind.valueOf(1, true) fails with java.lang.IllegalArgumentException will fix the implementation of valueOf(int refKind, boolean isInterface) for it to match its spec
@vicente-romero-oracle vicente-romero-oracle 16 hours ago Author Member

@mlchung I have updated the PR with another commit, thanks for your comments
@mlchung mlchung 12 hours ago Member

It may be better to create a new JBS issue to fix this bug as it may take more time to investigate.